### ISSUE TITLE
Would you contribute to ara if pull requests on GitHub were possible ?

### ISSUE THREAD

**Issue by dmsimard** [ansible-community/ara#205]
How odd that I would write this in a GitHub issue, you might say, and I wouldn't blame you :)

TL;DR: I'd like to lower the barrier to entry for contributing to ara and consider moving code review from [Gerrit](https://review.opendev.org/q/project:%255Erecordsansible/ara.*) to GitHub for the health and sustainability of the project. Read on to find out why I believe we should do this and let me know what you think.

I must preface this with the fact that I enjoy spending time on ara and the project isn't in any sort of danger.
Truth be told, ara doesn't have many regular contributors and my attempts at reaching out over the years have not been particularly successful but here's another one anyway ¯\\_(ツ)_/¯

If you look at the [contributor documentation](https://ara.readthedocs.io/en/latest/contributing.html), you'll find out that ara can't accept pull requests on GitHub and there's in fact a few hoops to jump through in order to send a patch, notably:

- creating an account on Ubuntu launchpad
- setting up gerrit and installing git-review
- proposing the patch to gerrit

I am forever grateful to those who took the time to send a patch (or more!) but I can't blame people for being turned off by the process and I'd like to address that.
For what it's worth, there are ways to synchronize GitHub PRs with gerrit and it works for [golang and sqlite](https://github.com/ansible-community/ara-infra/issues/4) but frankly, I'm not looking for **more** things to maintain -- we're talking about sustainability here :)

Otherwise, it turns out that running an open source project is a lot of work and it's not just about new features, fixing bugs or making things pretty.
In no particular order, it's things like:

- maintaining infrastructure ([website](https://ara.recordsansible.org/), [live demo](https://demo.recordsansible.org/), slack/irc bridge, etc)
- not letting the ssl certificates expire :(
- doing releases and packaging to [pypi](https://pypi.org/project/ara/), [fedora and epel8](https://koji.fedoraproject.org/koji/packageinfo?packageID=24394)
- building, testing and publishing container images to [DockerHub](https://hub.docker.com/r/recordsansible/ara-api) and [quay.io](https://quay.io/repository/recordsansible/ara-api)
- improving documentation
- talking and presenting about it
- helping users and keeping them updated
- writing [tweets](https://twitter.com/recordsansible) and [blog posts](https://ara.recordsansible.org/blog/)
- keeping CI jobs green
- getting nice stickers printed for fans
- stressing over trying to make the best decisions for the project and managing issues like this one!

A lot of this is automated, of course, but some isn't while some can't and I must acknowledge that there is a limit to my:

- time (dad ops is hard and ara has never been my full time job)
- skills (with a case of [imposter syndrome](https://en.wikipedia.org/wiki/Impostor_syndrome))
- interests (sorry, I don't enjoy javascript but have a lot of respect for people who do)

I'm happy to continue doing the above while improving ara to the best of my abilities but I feel that the project could be so much better with a little bit of help.

At a high level, there are four repositories:

- https://github.com/ansible-community/ara: Ansible modules and plugins, django/django-rest-framework API server and simple reporting UI, API and CLI clients
- https://github.com/ansible-community/ara-infra: website, live demo, infrastructure things
- https://github.com/ansible-community/ara-web: javascript web interface client for the API (currently broken)
- https://github.com/ansible-community/ara-collection: An Ansible collection of roles and playbooks to deploy the API server and ara-web

So I ask this question: Would you contribute to ara if pull requests on GitHub were possible ?

I could certainly use the help and I'm more than happy to answer questions.

Thanks !

P.S. I want to note that I am thankful to the OpenStack community and [OpenDev](https://opendev.org/) for hosting the project for many years. :heart:

**Comment by aorfanos** [comment]
Yes please! 😄

**Comment by karmab** [comment]
+1 !

**Comment by rsletten** [comment]
I would love to.

**Comment by usrGabriel** [comment]
I personally would not feel competent enough to contribute, but managing PRs on GitHub totally makes sense and makes the project a lot more accessible. So, I'm all for it for the day that I may feel competent enough to send PRs for this!

**Comment by ssbarnea** [comment]
Just do it! Is not as if you would have wanted to change its license or anything that could be seen as questionable.

**Comment by Thulium-Drake** [comment]
Hi @dmsimard ,

Good read, I also enjoy working with ARA (or on the Ansible collection ;-) ) and I feel life will become a lot easier on the project is on GH.

I feel the issue of time is one we all have to deal with, so I think it's for the better if the project can use the path of least resistance ;-)

**Comment by dmsimard** [comment]
Hey everyone and thanks for the feedback !

Many also commented on [reddit](https://www.reddit.com/r/ansible/comments/lhoojf/would_you_contribute_to_ara_if_pull_requests_on/) and [twitter](https://twitter.com/RecordsAnsible/status/1359905375894568965) about their interest.

I will start the process to move the code review over, which in practice means something along the lines of:
- [ ] Merging (or closing) any [pending changes in gerrit](https://review.opendev.org/#/q/project:%255Erecordsansible/ara.*+status:open)
- [ ] Letting opendev know about it, archiving and retiring repositories there
- [ ] Moving CI elsewhere (Moving to Zuul's Ansible would be almost drop-in since the project is already set up with Zuul)

In the meantime, feel free to submit PRs and we'll be able to merge them once CI is in place.

Looking forward to your contributions ! :tada:

**Comment by dmsimard** [comment]
I've merged and closed pending changes in Gerrit -- I've exported three to GitHub PRs because they are WIPs and I don't expect to land them before the change:
- WIP: UI: Add a host index https://github.com/ansible-community/ara/pull/206
- WIP: ui: add highlighted line on file line anchor https://github.com/ansible-community/ara/pull/207
- WIP: Add support for mod_wsgi https://github.com/ansible-community/ara-collection/pull/13

I've started looking at the CI changes and will write an email to openstack-discuss about the move before starting the archiving/retiring process.

**Comment by hemna** [comment]
Code reviews on github are absolutely terrible compared to gerrit.  This is a step in the wrong direction for open projects.

**Comment by dmsimard** [comment]
> Code reviews on github are absolutely terrible compared to gerrit. This is a step in the wrong direction for open projects.

@hemna The project has been hosted on Gerrit for the last five years and I don't recall seeing a patch from you but this issue isn't a debate about which tool is superior. You're likely to find as many people (if not more) complaining about Gerrit than people complaining about GitHub.

I'm tired of this.

**Comment by dmsimard** [comment]
I've made progress and will continue later.

The project is already hooked up to Ansible's Zuul: https://dashboard.zuul.ansible.com/t/ansible/builds?project=ansible-community%2Fara and I will be able to land a PR to add integration test jobs back in soon: https://github.com/ansible-community/ara/pull/208

As soon as we merge that PR, anyone can open a new pull request and it will be tested automatically.

I also proposed a patch to retire the projects on opendev: https://review.opendev.org/c/openstack/project-config/+/777675.
It needs a bit of work but it doesn't prevent us from accepting pull requests in the meantime.

**Comment by LaurentDumont** [comment]
Answered on the ML, but I think this a great idea! 👍
